### PR TITLE
Use updated kubeconfig library

### PR DESCRIPTION
**Pull request by rossf7**
Uses the updated kubeconfig library. See https://github.com/giantswarm/kubeconfig/pull/6.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -49,7 +49,7 @@ required = [
   name = "github.com/giantswarm/apiextensions"
 
 [[constraint]]
-  branch = "master"
+  branch = "remove-methods"
```
Comment (rossf7) [inline Gopkg.toml]: TODO

File: service/controller/app/v1/resource/status/create_test.go [hunk 2]
```diff
@@ -215,25 +213,9 @@ func Test_Resource_EnsureCreated(t *testing.T) {
...
-				}
-			}
-
 			c := Config{
-				G8sClient:  g8sClient,
-				K8sClient:  k8sfake.NewSimpleClientset(),
-				KubeConfig: kc,
-				Logger:     microloggertest.New(),
+				G8sClient: g8sClient,
+				Logger:    microloggertest.New(),
```
Comment (rossf7) [inline service/controller/app/v1/resource/status/create_test.go]: Simplify test.

File: service/controller/app/v1/resource/status/resource.go [hunk 3]
```diff
@@ -15,10 +13,8 @@ const (
 
 // Config represents the configuration used to create a new chartstatus resource.
 type Config struct {
-	G8sClient  versioned.Interface
-	K8sClient  kubernetes.Interface
-	KubeConfig kubeconfig.Interface
-	Logger     micrologger.Logger
+	G8sClient versioned.Interface
+	Logger    micrologger.Logger
```
Comment (rossf7) [inline service/controller/app/v1/resource/status/resource.go]: Simplify resource wiring.

File: service/controller/app/v1/resource_set.go [hunk 4]
```diff
@@ -159,12 +158,17 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 			return nil, microerror.Mask(err)
 		}
 
-		g8sClient, err := kubeConfig.NewG8sClientForApp(ctx, cr)
+		restConfig, err := kubeConfig.NewRESTConfigForApp(ctx, cr)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
```
Comment (rossf7) [inline service/controller/app/v1/resource_set.go]: This is the actual change. We get the REST config once and then add the clients to the context.

File: Gopkg.toml (hunk 1 above)
Comment (xh3b4sd) [inline Gopkg.toml]:
```suggestion
  branch = "master"
```

**Pull request merged**